### PR TITLE
chore(`infra.withArtifactCachingProxy`): use `pipeline-github-plugin` to retrieve PR labels

### PR DESCRIPTION
**Pull request by lemeurherve**
As suggested by @basil in https://github.com/jenkins-infra/pipeline-library/pull/552#discussion_r1152685801, this PR replaces the GitHub API `curl` call with a credentials by the use of the `pullRequest` object provided by https://github.com/jenkinsci/pipeline-github-plugin.

> > What would be the benefit(s) of switching to this method?
> 
> The current code uses a `@NonCPS` method to call `SCMSource.SourceByItem.findSource` against `currentBuild.rawBuild.parent`, and using `currentBuild.rawBuild` goes against Pipeline best practices (I can't remember the reference now, but I could look it up if this helps). Also, starting a new (CPS-based) `sh`/`bat` durable task step to invoke `curl` is more expensive than the simple Groovy code I showed above (which could be optimized even further with `@NonCPS`). So the alternative is cleaner and more efficient.

Tested in https://github.com/jenkinsci/jenkins-infra-test-plugin/pull/66 with and without the `skip-artifact-caching-proxy` label.

### CODE REVIEW

File: vars/infra.groovy [hunk 1]
```diff
@@ -121,23 +113,12 @@ Object withArtifactCachingProxy(boolean useArtifactCachingProxy = true, Closure
 
   // If the build concerns a pull request, check if there is "skip-artifact-caching-proxy" label applied in case the user doesn't want ACP
   if ((useArtifactCachingProxy) && (env.CHANGE_URL)) {
```
Comment (jglick) [inline vars/infra.groovy:115]: perhaps

```suggestion
  if (useArtifactCachingProxy && env.CHANGE_URL && pullRequest != null) {
```

File: vars/infra.groovy [hunk 2]
```diff
@@ -121,23 +113,12 @@ Object withArtifactCachingProxy(boolean useArtifactCachingProxy = true, Closure
...
-      if (isUnix()) {
-        prLabelsContainSkipACP = sh(script: 'curl --silent -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $GH_TOKEN" ' + pullrequestLabelsApiURL + ' | grep --ignore-case "skip-artifact-caching-proxy"', returnStatus: true) == 0
-      } else {
-        prLabelsContainSkipACP = bat(script: 'curl --silent -H "Accept: application/vnd.github+json" -H "Authorization: Bearer %GH_TOKEN%" ' + pullrequestLabelsApiURL + ' | findstr /i "skip-artifact-caching-proxy"', returnStatus: true) == 0
-      }
-      if (prLabelsContainSkipACP) {
-        echo "INFO: the label 'skip-artifact-caching-proxy' has been applied to the pull request, will use repo.jenkins-ci.org"
-        useArtifactCachingProxy = false
-      }
+    boolean prLabelsContainSkipACP = false
```
Comment (jglick) [inline vars/infra.groovy:116]: Forgotten?

```suggestion
```

**Review by jglick — DISMISSED** [review]

File: vars/infra.groovy (hunk 2 above)
Comment (lemeurherve) [reply vars/infra.groovy:116]: Good catch, thanks!

**Review by basil — APPROVED** [review]
Very nice!

**Pull request merged**